### PR TITLE
File uploader ics

### PR DESCRIPTION
**Pull request by stpCollabr8nLstn**
Closes carbon-design-system/carbon-addons-ics#6

Add optional `labelTooltip` prop
screen shot 2018-03-09 at 3 14 35 pm

`labelTooltip` prop accepts object structured as:
```js
const data = {
   text: 'some text',
   iconName: 'someIcon',
};
```

#### Changelog

**New**

- optional `labelTooltip` prop that takes an object with Tooltip text and iconName
- Add `FileUploader.js` from carbon-components-react
- Add `FileUploader-test.js` from carbon-components-react

**Changed**

- add scss styles for labelTooltip prop
- add FileUploader to `index.js`

### CODE REVIEW

File: src/tools/uniqueId.js [hunk 1]
```diff
@@ -0,0 +1,6 @@
+let lastId = 0;
```
Comment (j1mie) [inline src/tools/uniqueId.js]: Wondering if we can export this directly from carbon?

**Comment by j1mie** [comment]
Experiencing some weird display behaviour on tooltip

screen shot 2018-03-12 at 19 16 56

**Comment by stpCollabr8nLstn** [comment]
@j1mie It's coming from the Carbon internal FloatingMenu component. I can't override the styles because they're inline. I need to make it position: relative. Might need to submit a PR to Carbon.

File: src/components/FileUploader/FileUploader-story.js [hunk 2]
```diff
@@ -1,10 +1,20 @@
...
+import { FileUploaderButton } from 'carbon-components-react';
+import { FileUploader } from '../../index';
 import { withReadme } from 'storybook-readme';
 import readme from './README.md';
 
+const tooltipProps = {
+  text: 'only .jpg files at 500mb or less',
+  iconName: 'info',
+  menuOffset: {
+    top: -128,
```
Comment (j1mie) [inline src/components/FileUploader/FileUploader-story.js]: This is to accommodate the button height beneath?

File: src/components/FileUploader/README.md [hunk 3]
```diff
@@ -1,5 +1,25 @@
...
-import { FileUploader, FileUploaderButton } from 'carbon-components-react';
-```
\ No newline at end of file
+import { FileUploaderButton } from 'carbon-components-react';
+import { FileUploader } from 'carbon-addons-ics';
+```
+
+To enable the `labelTooltip` prop add a prop to your `FileUploader` like so:
+```js
+const data = {
```
Comment (j1mie) [inline src/components/FileUploader/README.md]: 👍

File: src/globals/internal/FloatingMenu.js [hunk 4]
```diff
@@ -0,0 +1,329 @@
+import warning from 'warning';
```
Comment (j1mie) [inline src/globals/internal/FloatingMenu.js]: Would like to test this out on other components using floating menu, as part of another task.

File: src/components/FileUploader/_file-uploader.scss [hunk 5]
```diff
@@ -10,6 +10,16 @@
     @include type-scale-item(a, false);
   }
 
+  .#{$prefix}--label-detail {
```
Comment (j1mie) [inline src/components/FileUploader/_file-uploader.scss]: This will eventually get its own component, I'm guessing!

Comment (stpCollabr8nLstn) [reply src/components/FileUploader/_file-uploader.scss]: Yup! This is sorta like label-tooltip-beta-1.0 ya know?

File: src/components/FileUploader/FileUploader-story.js (hunk 2 above)
Comment (stpCollabr8nLstn) [reply src/components/FileUploader/FileUploader-story.js]: yup. it places it here:
screen shot 2018-03-13 at 10 58 50 am

File: src/components/FileUploader/_file-uploader.scss (hunk 5 above)
Comment (j1mie) [reply src/components/FileUploader/_file-uploader.scss]: 👍

**Pull request merged**